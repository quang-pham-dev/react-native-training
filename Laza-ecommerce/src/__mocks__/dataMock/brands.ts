/* A data array. */

export const brand = {
  id: '0',
  name: 'Nike',
  logoUrl:
    'https://toppng.com/uploads/preview/ike-logo-png-hd-nike-logo-png-image-free-download-nike-logo-sv-11562865034fkhqkcoas6.png'
};

export const brands = [
  {
    id: '0',
    name: 'Nike',
    logoUrl:
      'https://toppng.com/uploads/preview/ike-logo-png-hd-nike-logo-png-image-free-download-nike-logo-sv-11562865034fkhqkcoas6.png'
  },
  {
    id: '1',
    name: 'Adidas',
    logoUrl:
      'https://www.pikpng.com/pngl/m/42-429866_adidas-logo-png-images-free-download-rh-pngimg.png'
  },
  {
    id: '2',
    name: 'Fila',
    logoUrl:
      'https://toppng.com/uploads/preview/fila-logo-vector-download-free-11574216707jcmvfu58fy.png'
  },
  {
    id: '3',
    name: 'Puma',
    logoUrl:
      'https://banner2.cleanpng.com/20180613/gsj/kisspng-tracksuit-puma-logo-brand-clothing-puma-logo-5b20dc6b232426.487874081528880235144.jpg'
  },
  {
    id: '4',
    name: 'Gucci',
    logoUrl: 'https://esquilo.io/png/thumb/XBwczcMncYeff65-Vector-Gucci-Logo-PNG-Free-Download.png'
  }
];
