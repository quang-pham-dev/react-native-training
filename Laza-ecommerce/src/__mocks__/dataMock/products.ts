import { ProductProps } from 'types/Products';

export const product: ProductProps = {
  id: '0',
  brandId: '1',
  name: 'Nike',
  price: 100,
  type: 'shoes',
  like: true,
  title: 'Nike Sportswear Club Fleece',
  source:
    'https://images.unsplash.com/photo-1542291026-7eec264c27ff?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470&q=80',
  description:
    "Nike is a multinaelichick and retired quarterback Jack Nicklaus. Nike's products include apparel, shoes, equipment, and accessories. Nike also owns and operates the majority of the world's sporting goods brands, including Adidas, Puma, and Fila.",
  reviewer: {
    id: '2',
    name: 'John Doe',
    date: '13, Sep, 2020',
    image:
      'https://images.unsplash.com/photo-1609505848912-b7c3b8b4beda?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=765&q=80',
  },
  imageReview: [
    {
      id: '0',
      image:
        'https://images.unsplash.com/photo-1499852848443-3004d6dc4cfc?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1374&q=80',
    },
    {
      id: '1',
      image:
        'https://images.unsplash.com/photo-1618898909019-010e4e234c55?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=687&q=80',
    },
    {
      id: '2',
      image:
        'https://images.unsplash.com/photo-1575537302964-96cd47c06b1b?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
    },
    {
      id: '3',
      image:
        'https://images.unsplash.com/photo-1518656306295-aa28b28b2504?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
    },
    {
      id: '4',
      image:
        'https://images.unsplash.com/photo-1609505848912-b7c3b8b4beda?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=765&q=80',
    },
    {
      id: '5',
      image:
        'https://images.unsplash.com/photo-1575537302964-96cd47c06b1b?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
    },
  ],
  rating: '4.5',
  comment:
    'lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam,adipisicing elit. Quisquam, quia.',
};

export const products: ProductProps[] = [
  {
    id: '0',
    brandId: '1',
    name: 'Nike',
    price: 100,
    type: 'shoes',
    like: true,
    title: 'Nike Sportswear Club Fleece',
    source:
      'https://images.unsplash.com/photo-1542291026-7eec264c27ff?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470&q=80',
    description:
      "Nike is a multinaelichick and retired quarterback Jack Nicklaus. Nike's products include apparel, shoes, equipment, and accessories. Nike also owns and operates the majority of the world's sporting goods brands, including Adidas, Puma, and Fila.",
    reviewer: {
      id: '2',
      name: 'John Doe',
      date: '13, Sep, 2020',
      image:
        'https://images.unsplash.com/photo-1609505848912-b7c3b8b4beda?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=765&q=80',
    },
    imageReview: [
      {
        id: '0',
        image:
          'https://images.unsplash.com/photo-1499852848443-3004d6dc4cfc?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1374&q=80',
      },
      {
        id: '1',
        image:
          'https://images.unsplash.com/photo-1618898909019-010e4e234c55?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=687&q=80',
      },
      {
        id: '2',
        image:
          'https://images.unsplash.com/photo-1575537302964-96cd47c06b1b?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
      },
      {
        id: '3',
        image:
          'https://images.unsplash.com/photo-1518656306295-aa28b28b2504?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
      },
      {
        id: '4',
        image:
          'https://images.unsplash.com/photo-1609505848912-b7c3b8b4beda?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=765&q=80',
      },
      {
        id: '5',
        image:
          'https://images.unsplash.com/photo-1575537302964-96cd47c06b1b?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
      },
    ],
    rating: '4.5',
    comment:
      'lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam,adipisicing elit. Quisquam, quia.',
  },
  {
    id: '0',
    brandId: '1',
    name: 'Nike',
    price: 100,
    type: 'shoes',
    like: true,
    title: 'Nike Sportswear Club Fleece',
    source:
      'https://images.unsplash.com/photo-1542291026-7eec264c27ff?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470&q=80',
    description:
      "Nike is a multinaelichick and retired quarterback Jack Nicklaus. Nike's products include apparel, shoes, equipment, and accessories. Nike also owns and operates the majority of the world's sporting goods brands, including Adidas, Puma, and Fila.",
    reviewer: {
      id: '2',
      name: 'John Doe',
      date: '13, Sep, 2020',
      image:
        'https://images.unsplash.com/photo-1609505848912-b7c3b8b4beda?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=765&q=80',
    },
    imageReview: [
      {
        id: '0',
        image:
          'https://images.unsplash.com/photo-1499852848443-3004d6dc4cfc?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1374&q=80',
      },
      {
        id: '1',
        image:
          'https://images.unsplash.com/photo-1618898909019-010e4e234c55?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=687&q=80',
      },
      {
        id: '2',
        image:
          'https://images.unsplash.com/photo-1575537302964-96cd47c06b1b?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
      },
      {
        id: '3',
        image:
          'https://images.unsplash.com/photo-1518656306295-aa28b28b2504?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
      },
      {
        id: '4',
        image:
          'https://images.unsplash.com/photo-1609505848912-b7c3b8b4beda?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=765&q=80',
      },
      {
        id: '5',
        image:
          'https://images.unsplash.com/photo-1575537302964-96cd47c06b1b?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
      },
    ],
    rating: '4.5',
    comment:
      'lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam,adipisicing elit. Quisquam, quia.',
  },
];
