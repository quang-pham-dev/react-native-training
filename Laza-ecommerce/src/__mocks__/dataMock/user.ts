export const user = {
  id: '621c4188dece66a40dad2951',
  avatar: 'https://picsum.photos/200',
  username: 'Quangpham',
  gender: 'male',
  email: 'admin@admin.com',
  phone: '+84 (919) 507231',
  address: '199 Bergen Street, Bowie, Oregon, 7933',
  role: 'admin',
  password: '123456'
};
