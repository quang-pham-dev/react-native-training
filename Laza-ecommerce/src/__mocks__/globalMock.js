import mockAsyncStorage from '@react-native-async-storage/async-storage/jest/async-storage-mock';

jest.mock('@react-native-community/async-storage', () => mockAsyncStorage);
