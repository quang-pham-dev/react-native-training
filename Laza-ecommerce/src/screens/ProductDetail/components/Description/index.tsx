import React, { memo } from 'react';
import { View } from 'react-native';

// Components
import Title from 'components/Title';
import MoreLessText from 'components/LessMoreText';

// Types
import { IProductDetailItemProps } from 'types/screens/ProductDetail';

// Styles
import styles from './styles';

const Description = ({ description }: IProductDetailItemProps) => {
  return (
    <View style={styles.descriptionContainer}>
      <Title.titleLeftSection titleName='Description' style={styles.textDescriptionTitle} />
      {Boolean(description) && (
        <MoreLessText styleShowMoreText={styles.textContent} numberOfLines={3}>
          {description}
        </MoreLessText>
      )}
    </View>
  );
};

export default memo(Description);
