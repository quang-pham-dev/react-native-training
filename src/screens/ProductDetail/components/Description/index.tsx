import React from 'react'

// Libs
import {s} from 'react-native-size-matters/extend'

// Components
import MoreLessText from '@components/LessMoreText'

// Styles
import PStyled from '@components/Paragraph/P.styles'

// Constants
import {PARAGRAPH_TYPE} from '@constants'

// Themes
import {Metrics} from '@themes'

type DescriptionProps = {
  description?: string
}

const Description = ({description}: DescriptionProps) => (
  <>
    <PStyled.Base
      mBottom={s(Metrics.margin.medium)}
      type={PARAGRAPH_TYPE.PRODUCT_DETAIL_TITLE_SECTION}>
      Description
    </PStyled.Base>
    {Boolean(description) && (
      <MoreLessText numberOfLines={3}>{description}</MoreLessText>
    )}
  </>
)

export default Description
