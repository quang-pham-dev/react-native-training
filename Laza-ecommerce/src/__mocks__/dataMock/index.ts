import { user } from './user';
import { brand, brands } from './brands';
import { product, products } from './products';

export { brand, brands, product, products, user };
