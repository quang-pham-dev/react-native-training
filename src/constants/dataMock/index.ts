import {user} from './user'
import {brand, brands} from './brands'

export {brand, brands, user}
