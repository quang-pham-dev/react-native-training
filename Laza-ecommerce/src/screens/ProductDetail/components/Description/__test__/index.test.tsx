import React from 'react';

// LIBS
import renderer from 'react-test-renderer';

// Components
import Description from 'screens/ProductDetail/components/Description';

// Mock data
import { product } from '__mocks__/dataMock';

describe('Product detail Description', () => {
  const tree = renderer.create(<Description description={product.description} />);

  test('should render correctly', () => {
    expect(tree).toMatchSnapshot();
  });
});
