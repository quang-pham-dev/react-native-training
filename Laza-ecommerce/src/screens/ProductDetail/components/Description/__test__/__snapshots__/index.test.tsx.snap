// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product detail Description should render correctly 1`] = `
<View
  style={
    Object {
      "marginTop": 20,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "textTransform": "none",
        },
        Object {
          "color": "#1D1E20",
          "fontFamily": "Inter_600SemiBold",
          "fontSize": 17,
          "lineHeight": 18,
          "marginBottom": 10,
        },
      ]
    }
  >
    Description
  </Text>
  <Text
    onTextLayout={[Function]}
    style={
      Array [
        Object {
          "fontFamily": "Inter_600SemiBold",
          "fontSize": 15,
          "lineHeight": 22,
        },
        Object {
          "color": "#8F959E",
          "fontFamily": "Inter_400Regular",
          "fontSize": 15,
          "lineHeight": 22,
        },
      ]
    }
  >
    Nike is a multinaelichick and retired quarterback Jack Nicklaus. Nike's products include apparel, shoes, equipment, and accessories. Nike also owns and operates the majority of the world's sporting goods brands, including Adidas, Puma, and Fila.
  </Text>
</View>
`;
